lib/fork.js

```diff
@@ -169,18 +169,23 @@ cluster.fork = function(env) {
       });
     }
   }
-  
+
+  function ondisconnect() {
+    var index = workers.indexOf(obj);
+    if (index >= 0) {
```

Comment:
Ah yeah, I made a mistake in not correctly knowing what `worksers.splice(-1, 1)` would do.

But why check for the index here if a new listener function is created for the `disconnect` event? Why not put this check inside the `onexit` function?